source/speech.py

```diff
@@ -2013,16 +2015,21 @@ class BaseCallbackCommand(SpeechCommand):
 	This command is never passed to synth drivers.
 	"""
 
+	@abstractmethod
 	def run(self):
 		"""Code to run when speech reaches this command.
+		Note that this method is executed in NVDA's main thread, therefore must return as soon as practically possible, otherwise it will block production of further speech and or other functionality in NVDA.
 		"""
 
 class CallbackCommand(BaseCallbackCommand):
 	"""Call a function when speech reaches this point.
 	"""
 
 	def __init__(self, callback):
-		self.run = callback
+		self._callback = callback
+
+	def run(self,*args, **kwargs):
+		return self._callback(*args,**kwargs)

```

Comment:
I wanted to suggest catching exceptions here, but then I realised that that's already done when executing a callback command. May be at a small entry to the docstring of BaseCallbackCommand.run saying that any exceptions caused by the callback are caught correctly, so the callback doesn't necessarily have to catch exceptions itself. In fact, any callback that catches exceptions itself and doesn't log these properly might obfuscate an issue. Just a suggestion though.